src/interpreter/MarkdownFileLocator.js

```diff
@@ -0,0 +1,38 @@
+const GenericFileLocator = require('./GenericFileLocator');
+
+const fs = require('fs');
+const path = require('path');
+
+class MarkdownFileLocator extends GenericFileLocator{
+    locateFiles(basePath, options){
+        options = typeof options === 'object' ? options : {
+            recursive: false
+        };
+
+        return this._readdir(basePath, options);
+    }
+
+    _readdir(basePath, options){
+        let self = this;
+        let files = fs.readdirSync(basePath, {withFileTypes: true});
+        let foundFiles = [];
+
+        files.forEach(function(file){
```

Comment:
Pour une fonction anonyme (comme ici), tu peux utiliser
```
files.forEach(file => {
   maFonctionCode
});
```

L'avantage de l"'arrow function", c'est qu'elle ne brise pas le scope de `this` dans le case où tu en as besoin, donc pas besoin de faire un `let self = this`.